middlewares/accesslog/logger.go

```diff
@@ -139,6 +140,22 @@ func (l *LogHandler) Close() error {
 	return l.file.Close()
 }
 
+// Rotate closes and reopens the log file to allow for rotation
+// by an external source.
+func (l *LogHandler) Rotate() error {
+	var err error
```

Comment:
Should suffice to be declared on line 150 only if we also scope `err` on line 147 to the if statement?